cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/model/ClusterModelStats.java

```diff
@@ -226,99 +228,73 @@ public String toString() {
+    Map<Resource, Double> minUtilizationPctByResource = new HashMap<>();
+    Map<Resource, Double> stDevUtilizationPctByResource = new HashMap<>();
     for (Resource resource : Resource.cachedValues()) {
-      double balanceUpperThreshold = (clusterModel.load().expectedUtilizationFor(resource) / clusterModel.capacityFor(resource))
-          * _balancingConstraint.resourceBalancePercentage(resource);
-      double balanceLowerThreshold = (clusterModel.load().expectedUtilizationFor(resource) / clusterModel.capacityFor(resource))
-          * Math.max(0, (2 - _balancingConstraint.resourceBalancePercentage(resource)));
-      // Average utilization for the resource.
-      double avgUtilization = clusterModel.load().expectedUtilizationFor(resource) / _numBrokers;
-      avgUtilizationByResource.put(resource, avgUtilization);
-
-      // Maximum, minimum, and standard deviation utilization for the resource.
-      double hottestBrokerUtilization = 0.0;
-      double coldestBrokerUtilization = Double.MAX_VALUE;
-      double variance = 0.0;
-      int numBalancedBrokers = 0;
+      double balanceUpperThreshold = _balancingConstraint.resourceBalancePercentage(resource);
+      double balanceLowerThreshold = Math.max(0, (2 - _balancingConstraint.resourceBalancePercentage(resource)));
+      List<Double> resourceUtilPcts = new ArrayList<>(numBrokers);
+      int i = 0;
       for (Broker broker : clusterModel.brokers()) {
-        double utilization = resource.isHostResource() ?
-            broker.host().load().expectedUtilizationFor(resource) : broker.load().expectedUtilizationFor(resource);
-        double balanceUpperLimit = resource.isHostResource() ?
-            broker.host().capacityFor(resource) * balanceUpperThreshold : broker.capacityFor(resource) * balanceUpperThreshold;
-        double balanceLowerLimit = resource.isHostResource() ?
-            broker.host().capacityFor(resource) * balanceLowerThreshold : broker.capacityFor(resource) * balanceLowerThreshold;
-        if (utilization <= balanceUpperLimit && utilization >= balanceLowerLimit) {
-          numBalancedBrokers++;
-        }
-        double brokerUtilization = broker.load().expectedUtilizationFor(resource);
-        hottestBrokerUtilization = Math.max(hottestBrokerUtilization, brokerUtilization);
-        coldestBrokerUtilization = Math.min(coldestBrokerUtilization, brokerUtilization);
-        variance += (Math.pow(brokerUtilization - avgUtilization, 2) / _numBrokers);
+        double utilization = resource.isHostResource() ? broker.host().load().expectedUtilizationFor(resource)
+                                                       : broker.load().expectedUtilizationFor(resource);
+        double capacity = resource.isHostResource() ? broker.host().capacityFor(resource)
+                                                    : broker.capacityFor(resource);
+        resourceUtilPcts.add(i++, utilization / capacity);
```

Comment:
`capacity` is `-1` for dead brokers. Hence, this will cause bad values in the computation of the following `avgUtilizationPct` and `stDevUtilizationPctByResource`.

Reply:
restricted to only consider alive brokers.